src/containers/cooperation-details/add-course-modal-modal/AddCourseTemplateModal.tsx

```diff
@@ -65,21 +61,29 @@ const AddCourseTemplateModal: FC<AddCourseTemplateModalProps> = ({
   }
 
   const getUserData = useCallback(
-    () => userService.getUserById(userId, userRole as UserRole),
+    () => userService.getUserByIdWithBaseService(userId, userRole as UserRole),
     [userId, userRole]
   )
 
-  const { loading: userLoading, response: user } =
-    useAxios<UserResponse | null>({
-      service: getUserData,
-      defaultResponse: null
-    })
+  const { isLoading: userLoading, data: userData = null } = useQuery({
+    queryFn: getUserData,
+    queryKey: ['user', userId],
+    options: {
+      staleTime: Infinity
+    }
+  })
 
-  const getCourses = useCallback(() => CourseService.getCourses(), [])
+  const getCourses = useCallback(
+    () => CourseService.getCoursesWithBaseService(),
+    []
+  )
```

Comment:
Remove this redundant wrapper, you can pass just reference
```suggestion
  const getCourses = useCallback(
    () => CourseService.getCoursesWithBaseService(),
    []
  )
```